helpers/splitWithSearchTerm.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+	convert string to array, separated by and including searchTerm
+	Ex:
+		Inputs: searchTerm='bon', str = 'bonJourbon'
+		Output: [bon, Jour, bon]
+*/
+export const splitWithSearchTerm = (str: string, searchTerm: string) => {
```

Comment:
This could potentially be replaced by a simpler code using split with a regex with a capturing group to include the search term and case insensitive flag (i). For example:

```js
const splitWithSearchTerm = (str, searchTerm) => {
  const regex = new RegExp(`(${searchTerm})`, 'i')
  return str.split(regex).filter(s => s !== '')
  // filter to remove the empty strings that appear if searchTerm is at the start or end
}
```

Results in:

```js
splitWithSearchTerm('bonJourbon', 'bon') = [ 'bon', 'Jour', 'bon' ]
splitWithSearchTerm('hiNgaDiNGADuurgen', 'dINga') = [ 'hiNga', 'DiNGA', 'Duurgen' ]
splitWithSearchTerm('hiNgaDiNGADuurgen', 'DINGA') = [ 'hiNga', 'DiNGA', 'Duurgen' ]
splitWithSearchTerm('hiNgaDiNGADuurgen', 'dinga') = [ 'hiNga', 'DiNGA', 'Duurgen' ]
```